lib/CTownHandler.cpp

```diff
@@ -718,9 +718,12 @@ CFaction * CTownHandler::loadFromJson(const JsonNode &source, const std::string
 	faction->creatureBg120 = source["creatureBackground"]["120px"].String();
 	faction->creatureBg130 = source["creatureBackground"]["130px"].String();
 
+	auto nativeTerrain = source["nativeTerrain"];
+
+	faction->nativeTerrain = nativeTerrain.isNull()
+		? ETerrainType(defaultFactionTerrain)

```

Comment:
Is this cast actually needed?

Reply:
`ETerrainType(defaultFactionTerrain)` in this case is same as `defaultFactionTerrain` 

Firstly, `ETerrainType(EETerrainType _num = WRONG) : num(_num)`
constructor will be called, then `ETerrainType& operator=(const ETerrainType& other)`

In the first case it's clear at a glance, that lvalue has  ETerrainType type, not  EETerrainType

ID_LIKE_CLASS_COMMON(ETerrainType, EETerrainType) macros does not
produce operator= which takes EETerrainType as argument.

But, if I add such operator= , one less object will be constructed